server/utils/retrieveQuestionResponseFromApplicationOrAssessment.ts

```diff
@@ -0,0 +1,62 @@
+import {
+  ApprovedPremisesApplication as Application,
+  ApprovedPremisesAssessment as Assessment,
+} from '@approved-premises/api'
+import { TasklistPageInterface } from '../form-pages/tasklistPage'
+import { getPageDataFromApplicationOrAssessment, getPageName } from '../form-pages/utils'
+import { SessionDataError } from './errors'
+import { camelCase } from './utils'
+
+/**
+ * Retrieves response for a given question from the application object or throws an error if it does not exist.
+ * @param application the application to fetch the response from.
+ * @param task the task to retrieve the response for.
+ * @param page the page that we need the response for in camelCase.
+ * @param {string} question [question=page] the page that we need the response for. Defaults to the value of `page`.
+ * @returns the response for the given task/page/question.
+ */
+export const retrieveQuestionResponseFromApplicationOrAssessment = <T>(
+  applicationOrAssessment: Application | Assessment,
+  Page: unknown,
+  question?: string,
+) => {
+  const pageData = getPageDataFromApplicationOrAssessment(Page as TasklistPageInterface, applicationOrAssessment)
+  const pageName = getPageName(Page)
+  const q = question || camelCase(pageName)
+
+  if (!pageData) {
+    throw new SessionDataError(`Question ${q} was not found in the session`)
+  }
+
+  const response = pageData[q]
+
+  if (!response) {
+    throw new SessionDataError(`Question ${q} was not found in the session`)
+  }
+
+  return response as T
+}
+
+/**
+ * Retrieves response for a given question from the application object or returns undefined if it does not exist.
+ * @param application the application to fetch the response from.
```

Comment:
These are also outdated

Reply:
I wonder if there is something we can do to ensure the jsdocs are correct at a type level for the function they document. 
Maybe [eslint-plugin-jsdoc](https://www.npmjs.com/package/eslint-plugin-jsdoc) can help?